src/main/java/org/sagebionetworks/web/client/widget/entity/controller/EntityActionControllerImpl.java

```diff
@@ -147,6 +147,10 @@ public void configure(ActionMenuWidget actionMenu,
 		actionMenu.addControllerWidget(uploader.asWidget());
 		if (!isUserAuthenticated) {
 			actionMenu.setToolsButtonVisible(false);
+			if (permissions.getCanPublicRead()) {

```

Comment:
The smoke testing seemed good when I checked against a public project then a public file.